src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/ReferenceDataService.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.hmcts.reform.sandl.snlevents.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sandl.snlevents.interfaces.SimpleDictionarySettable;
+import uk.gov.hmcts.reform.sandl.snlevents.model.response.CaseTypeWithHearingTypesResponse;
+import uk.gov.hmcts.reform.sandl.snlevents.model.response.SimpleDictionaryData;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.CaseTypeRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.HearingTypeRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.RoomTypeRepository;
+import uk.gov.hmcts.reform.sandl.snlevents.repository.db.SessionTypeRepository;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+public class ReferenceDataService {
+
+    private final Function<SimpleDictionarySettable, SimpleDictionarySettable> toSimpleDictionaryData =
+        (SimpleDictionarySettable sds) -> {
+            SimpleDictionarySettable response = new SimpleDictionaryData(sds.getCode(), sds.getDescription());
+
+            return response;
+        };
+    @Autowired
+    CaseTypeRepository caseTypeRepository;
+    @Autowired
+    SessionTypeRepository sessionTypeRepository;
+    @Autowired
+    HearingTypeRepository hearingTypeRepository;
+    @Autowired
+    RoomTypeRepository roomTypeRepository;
+
+    public List<CaseTypeWithHearingTypesResponse> getCaseTypes() {
+        return caseTypeRepository
+            .findAll()
+            .stream()
+            .map(caseType -> {
+                CaseTypeWithHearingTypesResponse mappedTo = new CaseTypeWithHearingTypesResponse();
```

Comment:
`CaseTypeWithHearingTypesResponse` should has constructor that take code and desc, so you can set them here and remove setters from lines 49 & 50

Reply:
surprise surprise, it does not ;) most probably a missing Lombok annotation to use parent fields. WIll have a look